src/main/java/mate/academy/springbootstore/service/BookServiceImpl.java

```diff
@@ -18,22 +18,52 @@ public class BookServiceImpl implements BookService {
 
     @Override
     public List<BookDto> getAll() {
-        return bookRepository.getAll().stream()
+        return bookRepository.findAll().stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
 
     @Override
     public BookDto getBookById(Long id) {
-        Book book = bookRepository.getBookById(id)
+        Book book = bookRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by Id: " + id));
         return bookMapper.toDto(book);
     }
 
     @Override
     public BookDto createBook(CreateBookRequestDto requestDto) {
         Book savedBook = bookMapper.toModel(requestDto);
-        bookRepository.createBook(savedBook);
+        bookRepository.save(savedBook);
         return bookMapper.toDto(savedBook);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't delete book. Id not found: " + id);
+        }
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(BookDto bookDto) {
+        Long id = bookDto.getId();
+        if (id == null) {
+            throw new IllegalArgumentException("Book ID must not be null for update");
+        }
+
+        Book bookFromDb = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't update book. Id not found: " + id));
+
+        bookFromDb.setTitle(bookDto.getTitle());
+        bookFromDb.setAuthor(bookDto.getAuthor());
+        bookFromDb.setPrice(bookDto.getPrice());
+        bookFromDb.setIsbn(bookDto.getIsbn());
+        bookFromDb.setDescription(bookDto.getDescription());
+        bookFromDb.setCoverImage(bookDto.getCoverImage());
+
+
+        Book updatedBook = bookRepository.save(bookFromDb);
+        return bookMapper.toDto(updatedBook);

```

Comment:
create method in mapper 
`void updateBookFromDto(CreateBookRequestDto book, @MappingTarget Book entity);`
```suggestion
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(book);
```